apollo-federation/src/sources/connect/models.rs

```diff
@@ -365,29 +365,20 @@ fn determine_entity_resolver(
 fn extract_header_references(
     variable_references: HashSet<VariableReference<Namespace>>,
 ) -> HashSet<String> {
-    let headers: HashSet<String> = variable_references
+    variable_references
         .iter()
-        .filter_map(|var_ref| {
+        .flat_map(|var_ref| {
             if var_ref.namespace.namespace != Namespace::Request
                 && var_ref.namespace.namespace != Namespace::Response
             {
-                return None;
-            }
-
-            // We only care if the path references starts with "headers"
-            if var_ref
-                .path
-                .first()
-                .is_none_or(|path| path.part != "headers")
-            {
-                return None;
+                vec![]
+            } else if let Some(headers_subselection) = var_ref.selection.get("headers") {
+                headers_subselection.keys().cloned().collect()
+            } else {
+                vec![]
```

Comment:
generally avoid `vec![]` as it is not const, prefer `Vec::new()` as it is const, https://doc.rust-lang.org/std/vec/struct.Vec.html#method.new

Reply:
https://github.com/apollographql/router/pull/7419/commits/d7e8d5e70b57f1d94ea5d117506c2cc4b235b031